### PR TITLE
feat(specs): add new events type for insights

### PR DESCRIPTION
**Pull request by millotp**
## 🧭 What and Why

🎟 JIRA Ticket: [APIC-716](https://algolia.atlassian.net/browse/APIC-716)

Add the 4 new types of events for insights:
- `AddedToCartObjectIDs`
- `AddedToCartObjectIDsAfterSearch`
- `PurchasedObjectIDs`
- `PurchasedObjectIDsAfterSearch`

[APIC-716]: https://algolia.atlassian.net/browse/APIC-716?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: specs/insights/common/enums.yml [hunk 1]
```diff
@@ -1,3 +1,7 @@
 eventType:
   type: string
   enum: [click, conversion, view]
+
+conversionEventSubType:
+  type: string
+  enum: [addToCart, purchase]
```
Comment (shortcuts) [inline specs/insights/common/enums.yml:7]: is this used?

File: specs/insights/common/schemas/EventsItems.yml [hunk 2]
```diff
@@ -7,3 +7,7 @@ oneOf:
   - $ref: './ConvertedFilters.yml'
   - $ref: './ViewedObjectIDs.yml'
   - $ref: './ViewedFilters.yml'
+  - $ref: './AddedToCartObjectIDsAfterSearch.yml'
+  - $ref: './AddedToCartObjectIDs.yml'
+  - $ref: './PurchasedObjectIDs.yml'
+  - $ref: './PurchasedObjectIDsAfterSearch.yml'
```
Comment (shortcuts) [inline specs/insights/common/schemas/EventsItems.yml:13]: I feel like some of those objects could benefits from allOfs but might be harder to maintain, up to you

**Review by shortcuts — COMMENTED** [review]
looks great on the generation side!

File: specs/insights/common/schemas/EventsItems.yml (hunk 2 above)
Comment (millotp) [reply specs/insights/common/schemas/EventsItems.yml:13]: yeah only the eventSubType changes, I can give it a try

File: specs/insights/common/enums.yml (hunk 1 above)
Comment (millotp) [reply specs/insights/common/enums.yml:7]: no but it's also exported in `search-insights.js`, I thought some people could use it.
Actually it's not event generated, I can remove the file.

Comment (shortcuts) [reply specs/insights/common/enums.yml]: yup I think we ignore unused models at generation time but if this is used in the lib maybe it's worth adding it to the client directly

File: specs/insights/common/schemas/EventsItems.yml (hunk 2 above)
Comment (millotp) [reply specs/insights/common/schemas/EventsItems.yml:13]: I don't sure the generator is mature enough, let's postpone that to v7, here is not working properly.

File: specs/insights/common/schemas/EventAttributes.yml [hunk 3]
```diff
@@ -81,3 +82,68 @@ positions:
...
+      description: The quantity of the purchased or added-to-cart item. The total value of a purchase is the sum of `quantity` multiplied with the `price` for each purchased item.
+    discount:
+      $ref: '#/discount'
+
+objectDataList:
+  type: array
+  description: |
+    Extra information about the records involved in the event—for example, to add price and quantities of purchased products.
+
+    If provided, must be the same length as objectIDs.
```
Comment (jkaho) [inline specs/insights/common/schemas/EventAttributes.yml:118]:
```suggestion
    If provided, must be the same length as `objectIDs`.
```

File: specs/insights/common/schemas/EventAttributes.yml [hunk 4]
```diff
@@ -81,3 +82,68 @@ positions:
...
+      description: The quantity of the purchased or added-to-cart item. The total value of a purchase is the sum of `quantity` multiplied with the `price` for each purchased item.
+    discount:
+      $ref: '#/discount'
+
+objectDataAfterSearchList:
+  type: array
+  description: |
+    Extra information about the records involved in the event—for example, to add price and quantities of purchased products.
+
+    If provided, must be the same length as objectIDs.
```
Comment (jkaho) [inline specs/insights/common/schemas/EventAttributes.yml:143]:
```suggestion
    If provided, must be the same length as `objectIDs`.
```

File: specs/insights/common/schemas/PurchasedObjectIDs.yml [hunk 5]
```diff
@@ -0,0 +1,30 @@
+type: object
+title: Purchased object IDs
+description: Use this event to track when users make a purchase unrelated to a previous Algolia request. For example, if you don't use Algolia to build your category pages, use `purchasedObjectIDs`.
```
Comment (jkaho) [inline specs/insights/common/schemas/PurchasedObjectIDs.yml:3]:
```suggestion
description: Use this event to track when users make a purchase unrelated to a previous Algolia request. For example, if you don't use Algolia to build your category pages, use this event.
```

File: specs/insights/common/schemas/PurchasedObjectIDsAfterSearch.yml [hunk 6]
```diff
@@ -0,0 +1,35 @@
+type: object
+title: Purchased object IDs after search
+description: |
+  Use this event to track when users make a purchase after a previous Algolia request.
+  If you use Algolia to build your category pages, you'll also use the `purchasedObjectIDsAfterSearch` event.
```
Comment (jkaho) [inline specs/insights/common/schemas/PurchasedObjectIDsAfterSearch.yml:5]:
```suggestion
  If you're building your category pages with Algolia, you'll also use this event.
```

**Review by jkaho — COMMENTED** [review]
LGTM, just some minor suggestions

File: specs/insights/common/schemas/AddedToCartObjectIDs.yml [hunk 7]
```diff
@@ -0,0 +1,30 @@
+type: object
+title: Added to cart object IDs
+description: Use this event to track when users add items to their shopping cart unrelated to a previous Algolia request. For example, if you don't use Algolia to build your category pages, use `addedToCartObjectIDs`.
+additionalProperties: false
+properties:
+  eventName:
+    $ref: './EventAttributes.yml#/eventName'
+  eventType:
+    $ref: './ConversionEvent.yml'
+  eventSubType:
```
[the remark this answers is not in the document]
Comment (millotp) [reply specs/insights/common/schemas/AddedToCartObjectIDs.yml:10]: oops yes that's a mistake, good catch 😃

**Comment by millotp** [comment]
Thanks for the review @jkaho, it should be resolved by e7d01523f8e97ced6bb1abe18e7b6e3d98bd170e 😃

File: specs/insights/common/schemas/PurchasedObjectIDs.yml [hunk 8]
```diff
@@ -0,0 +1,30 @@
+type: object
+title: Purchased object IDs
+description: Use this event to track when users make a purchase unrelated to a previous Algolia request. For example, if you don't use Algolia to build your category pages, use this event.
```
Comment (jkaho) [inline specs/insights/common/schemas/PurchasedObjectIDs.yml:3]: Sorry, missed this!
```suggestion
description: Use this event to track when users make a purchase unrelated to a previous Algolia request. For example, if you don't use Algolia to build your category pages, use this event.

To track purchase events related to Algolia requests, use the "Purchased object IDs after search" event.
```

Comment (jkaho) [reply specs/insights/common/schemas/PurchasedObjectIDs.yml:3]: (Might need to update the formatting 😅)

**Review by jkaho — APPROVED** [review]

**Review by shortcuts — APPROVED** [review]
gg!!

**Pull request merged**